eth_worker/sql_persistence/interface.py

```diff
@@ -503,8 +504,9 @@ def add_block_range(self, start, end, filter_id):
 
     def set_block_range_status(self, start, end, status):
```

Comment:
Possible statuses: PENDING, SUCCESS, FAILED.
`add_block_range` takes a filter param, so assuming filters can return txs in the same block, there will be multiple rows with the same block number, but different filter_ids.
`set_block_range_status` does not filter by filter_id, so the multiple rows being set for a block_number may include FAILED statuses of other filters?

Reply:
Wow, good catch! Fixed